### ISSUE TITLE
Hyperopt runs, but the epoches dont trade

### ISSUE THREAD

**Issue by Elmata2** [freqtrade/freqtrade#6647]
## Describe your environment

  * Operating system: ____
  * Python Version: _____ (`python -V`)
  * CCXT version: _____ (`pip freeze | grep ccxt`)
  * Freqtrade Version: ____ (` `docker-compose run --rm freqtrade -V` for Freqtrade running in docker)
  
## Your question
He man, 
the hyperopt runs epoches but it doesnt trade? this is the command i am running in cmd docker-compose run freqtrade hyperopt --config /freqtrade/user_data/config.json --hyperopt-loss SharpeHyperOptLoss --strategy BBRSIJortStrategy -e 500 --spaces all

i get this 
```
2022-04-04 16:24:57,222 - freqtrade.optimize.backtesting - INFO - Loading data from 2022-02-22 21:45:00 up to 2022-04-03 19:30:00 (39 days).
2022-04-04 16:24:57,222 - freqtrade.configuration.timerange - WARNING - Moving start-date by 25 candles to account for startup time.
2022-04-04 16:24:57,223 - freqtrade.optimize.hyperopt - INFO - Dataload complete. Calculating indicators
2022-04-04 16:24:58,135 - freqtrade.optimize.hyperopt - INFO - Hyperopting with data from 2022-02-23 04:00:00 up to 2022-04-03 19:30:00 (39 days)..
2022-04-04 16:24:58,501 - freqtrade.exchange.exchange - INFO - Closing async ccxt session.
2022-04-04 16:24:58,525 - freqtrade.optimize.hyperopt - INFO - Found 4 CPU cores. Let's make them scream!
2022-04-04 16:24:58,525 - freqtrade.optimize.hyperopt - INFO - Number of parallel jobs set as: -1
2022-04-04 16:24:58,527 - freqtrade.optimize.hyperopt - INFO - Using estimator ET.
2022-04-04 16:24:58,548 - freqtrade.optimize.hyperopt - INFO - Effective number of parallel workers used: 4
 [Epoch 500 of 500 (100%)] ||                                                                                                                                            | [Time:  0:16:50, Elapsed Time: 0:16:50]
2022-04-04 16:41:56,406 - freqtrade.optimize.hyperopt - INFO - 500 epochs saved to '/freqtrade/user_data/hyperopt_results/strategy_BBRSIJortStrategy_2022-04-04_16-24-56.fthypt'.
No epochs evaluated yet, no best result."
```

This is the strat i am using
```
# Start hyperopt with the following command:
# freqtrade hyperopt --config config.json --hyperopt-loss SharpeHyperOptLoss --strategy RsiStrat -e 500 --spaces  buy sell --random-state 8711

# --- Do not remove these libs ---
import numpy as np  # noqa
import pandas as pd  # noqa
from functools import reduce
from pandas import DataFrame

from freqtrade.strategy import (BooleanParameter, CategoricalParameter, DecimalParameter,IStrategy, IntParameter)

# --- Add your lib to import here ---
import talib.abstract as ta
import freqtrade.vendor.qtpylib.indicators as qtpylib

# --- Generic strategy settings ---

class BBRSIJortStrategy(IStrategy):
    INTERFACE_VERSION = 2
    # Determine timeframe and # of candles before strategysignals becomes valid
    timeframe = '15m'
    startup_candle_count: int = 25
    # Determine roi take profit and stop loss points
    minimal_roi = {
        "60":  0.01,
        "30":  0.03,
        "20":  0.04,
        "0":  0.05
    }

    stoploss = -0.10
    trailing_stop = False
    use_sell_signal = True
    sell_profit_only = False
    sell_profit_offset = 0.0
    ignore_roi_if_buy_signal = False

    # Define the parameter spaces
    cooldown_lookback = IntParameter(2, 48, default=5, space="protection", optimize=True)
    stop_duration = IntParameter(12, 200, default=5, space="protection", optimize=True)
    use_stop_protection = BooleanParameter(default=True, space="protection", optimize=True)


    @property
    def protections(self):
        prot = []

        prot.append({
            "method": "CooldownPeriod",
            "stop_duration_candles": self.cooldown_lookback.value
        })
        if self.use_stop_protection.value:
            prot.append({
                "method": "StoplossGuard",
                "lookback_period_candles": 24 * 3,
                "trade_limit": 4,
                "stop_duration_candles": self.stop_duration.value,
                "only_per_pair": False
            })

        return prot

# --- Define spaces for the indicators ---

    buy_rsi= IntParameter(5, 50,default=30, space="buy")
    buy_rsi_enabled = CategoricalParameter([True, False],default=True, space="buy")
    buy_trigger = CategoricalParameter(['tr_bb_lower_1sd', 'tr_bb_lower_2sd', 'tr_bb_lower_3sd', 'tr_bb_lower_4sd'], default='tr_bb_lower_1sd' , space="buy")

    sell_rsi = IntParameter(30, 100, default=50, space="sell")
    sell_rsi_enabled = CategoricalParameter([True, False], default=True, space="sell")
    sell_trigger = CategoricalParameter(['sell_tr_bb_lower_1sd','sell_tr_bb_mid_1sd','sell_tr_bb_upper_1sd'],default='sell_tr_bb_lower_1sd',space="sell")


# --- Used indicators of strategy code ----

    def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        # rsi's for buying and selling

        for val in self.buy_rsi.range:
            dataframe['buy_rsi_{val}'] = ta.RSI(dataframe, timeperiod=val)
        for val in self.sell_rsi.range:
            dataframe[f'sell_rsi_{val}'] = ta.RSI(dataframe, timeperiod=val)


        # Bollinger bands
        bollinger_1sd = qtpylib.bollinger_bands(qtpylib.typical_price(dataframe), window=20, stds=1)
        dataframe['bb_lowerband_1sd'] = bollinger_1sd['lower']
        dataframe['bb_middleband_1sd'] = bollinger_1sd['mid']
        dataframe['bb_upperband_1sd'] = bollinger_1sd['upper']

        bollinger_2sd = qtpylib.bollinger_bands(qtpylib.typical_price(dataframe), window=20, stds=2)
        dataframe['bb_lowerband_2sd'] = bollinger_2sd['lower']
        dataframe['bb_middleband_2sd'] = bollinger_2sd['mid']
        dataframe['bb_upperband_2sd'] = bollinger_2sd['upper']

        bollinger_3sd = qtpylib.bollinger_bands(qtpylib.typical_price(dataframe), window=20, stds=3)
        dataframe['bb_lowerband_3sd'] = bollinger_3sd['lower']
        dataframe['bb_middleband_3sd'] = bollinger_3sd['mid']
        dataframe['bb_upperband_3sd'] = bollinger_3sd['upper']

        bollinger_4sd = qtpylib.bollinger_bands(qtpylib.typical_price(dataframe), window=20, stds=4)
        dataframe['bb_lowerband_4sd'] = bollinger_4sd['lower']
        dataframe['bb_middleband_4sd'] = bollinger_4sd['mid']
        dataframe['bb_upperband_4sd'] = bollinger_4sd['upper']

        return dataframe

# --- Buy settings ---

    def populate_buy_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        
        conditions = []
        if self.buy_trigger == 'tr_bb_lower_1sd':
            conditions.append(dataframe['close'] < dataframe['bb_lowerband_1sd'])
        if self.buy_trigger.value == 'tr_bb_lower_2sd':
            conditions.append(dataframe['close'] < dataframe['bb_lowerband_2sd'])
        if self.buy_trigger.value == 'tr_bb_lower_3sd':
            conditions.append(dataframe['close'] < dataframe['bb_lowerband_3sd'])
        if self.buy_trigger == 'tr_bb_lower_4sd':
            conditions.append(dataframe['close'] < dataframe['bb_lowerband_4sd']
            )

        if conditions:
               dataframe.loc[
               reduce(lambda x, y: x & y, conditions),
               'buy'] = 1

        return dataframe    

# --- sell settings ---
    def populate_sell_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        conditions = []
        # TRIGGERS
        
        if self.sell_trigger == 'sell_tr_bb_lower_1sd':
                    conditions.append(dataframe['close'] > dataframe['bb_lowerband_1sd'])
        if self.sell_trigger == 'sell_tr_bb_mid_1sd':
                    conditions.append(dataframe['close'] > dataframe['bb_middleband_1sd'])
        if self.sell_trigger == 'sell_tr_bb_upper_1sd':
                    conditions.append(dataframe['close'] > dataframe['bb_upperband_1sd'])
        
        # Check that the candle had volume
        conditions.append(dataframe['volume'] > 0)

        if conditions:
            dataframe.loc[
                reduce(lambda x, y: x & y, conditions),
                    'sell'] = 1
            return dataframe

        return populate_sell_trend

           
```
*Ask the question you have not been able to find an answer in the [Documentation](https://www.freqtrade.io/en/latest/)*
"Do you know a solution?

**Comment by xmatthias** [comment]
at a quick glance i'd say that the "if" conditions are wrong.
`if self.buy_trigger == 'tr_bb_lower_1sd':` will always evaluate to false - you'll have to use `if self.buy_trigger.value == 'tr_bb_lower_1sd':`

**Comment by Elmata2** [comment]
Thats why you are the master. Your quick glance was exactly what was wrong, it works fine now
